src/modules/admin/components/formRules/FormRule.tsx

```diff
@@ -143,4 +122,57 @@ const FormRule: React.FC<Props> = ({ rule, formRole, actionButtons }) => {
   );
 };
 
-export default FormRule;
+interface FormRuleProps {
+  rule: FormRuleFieldsFragment;
+  formRole?: FormRole;
+  actionButtons?: ReactNode;
+}
+export const FormRule: React.FC<FormRuleProps> = ({
+  rule,
+  formRole,
+  actionButtons,
+}) => {
+  return (
+    <BaseFormRule
+      project={rule.project?.projectName}
+      projectType={
+        rule.projectType ? HmisEnums.ProjectType[rule.projectType] : ''
```

Comment:
What's the utility of pulling out the display logic into the parent components (FormRule/ProjectConfigFormRule), as opposed to having BaseFormRule be responsible for displaying values?

If it was a TS concern, an option is:
```
interface BaseFormRuleProps {
  projectType?: ProjectType | null;
}
// in BaseFormRule
label={`Project Type is ${HmisEnums.ProjectType[projectType]}`}

// passing the prop:
projectType={rule.projectType}
```

If `BaseFormRuleProps` continues to accept all strings, it might be worth renaming them for clarity. Eg `projectType=>projectTypeDisplay` `project=>projectName` etc. to clarify.

Reply:
Hm, yes I agree, this refactor is better. It can just accept a typed `ProjectType`, `DataCollectedAbout`, and `Funder`. For organization name, project name, and service type/category name, I think it must accept strings since those are the values that the `NewFormRuleDialog` has access to, not the full project/org object. I will rename those params for clarity as you suggested.